static/app/components/smartSearchBar/utils.tsx

```diff
@@ -121,14 +125,34 @@ export function createSearchGroups(
   }
 
   if (queryCharsLeft || queryCharsLeft === 0) {
+    searchItems = searchItems.flatMap(item => {
```

Comment:
This function is getting pretty large, could use a little TLC. Would love it some of this logic was extracted into functions with descriptive names.